django/db/models/fields/reverse_related.py

```diff
@@ -115,6 +115,28 @@ def __repr__(self):
             self.related_model._meta.model_name,
         )
 
+    @property
+    def identity(self):
+        return (
+            self.field,
+            self.model,
+            self.related_name,
+            self.related_query_name,
+            tuple(sorted(self.limit_choices_to.items()))
+            if isinstance(self.limit_choices_to, dict)
+            else self.limit_choices_to,
+            self.parent_link,
+            self.on_delete,
+            self.symmetrical,
+            self.multiple,
+        )
+
+    def __eq__(self, other):
+        return isinstance(other, self.__class__) and self.identity == other.identity
```

Comment:
We should return `NotImplemented` if objects are comparable:
```python
if not isinstance(other, self.__class__):
    return NotImplemented
... 
```

Reply:
Shouldn't the equality operator return false if we compare two objects of different types?

Changing this starts to throw errors in the tests:
```
======================================================================
ERROR: test_filter_reverse_fk (queryset_pickle.tests.PickleabilityTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python3.8/unittest/case.py", line 60, in testPartExecutor
    yield
  File "/usr/lib/python3.8/unittest/case.py", line 676, in run
    self._callTestMethod(testMethod)
  File "/usr/lib/python3.8/unittest/case.py", line 633, in _callTestMethod
    method()
  File "mypath/django/tests/queryset_pickle/tests.py", line 39, in test_filter_reverse_fk
    self.assert_pickles(Group.objects.filter(event=1))
  File "mypath/django/django/db/models/manager.py", line 85, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "mypath/django/django/db/models/query.py", line 932, in filter
    return self._filter_or_exclude(False, args, kwargs)
  File "mypath/django/django/db/models/query.py", line 952, in _filter_or_exclude
    clone._filter_or_exclude_inplace(negate, args, kwargs)
  File "mypath/django/django/db/models/query.py", line 959, in _filter_or_exclude_inplace
    self._query.add_q(Q(*args, **kwargs))
  File "mypath/django/django/db/models/sql/query.py", line 1376, in add_q
    clause, _ = self._add_q(q_object, self.used_aliases)
  File "mypath/django/django/db/models/sql/query.py", line 1395, in _add_q
    child_clause, needed_inner = self.build_filter(
  File "mypath/django/django/db/models/sql/query.py", line 1324, in build_filter
    col = self._get_col(targets[0], join_info.final_field, alias)
  File "mypath/django/django/db/models/sql/query.py", line 362, in _get_col
    return target.get_col(alias, field)
  File "mypath/django/django/db/models/fields/__init__.py", line 398, in get_col
    if alias != self.model._meta.db_table or output_field != self:
  File "mypath/django/django/db/models/fields/reverse_related.py", line 138, in __eq__
    raise NotImplementedError
NotImplementedError
```

where `self` is `<ManyToOneRel: queryset_pickle.event>`
and `other` is a `queryset_pickle.Event.id` (of type: `<class 'django.db.models.fields.AutoField'>`

Which seems to make sense, that they should be comparable but obviously not equal 🤔

Reply:
Not `raise NotImplementedError` but `return NotImplemented` (see [docs](https://docs.python.org/3/library/constants.html#NotImplemented)).

Reply:
TIL :) Thanks! Made the change ✅